### DISCUSSION TITLE
Change the sequence rendering in the sequence panel

### DISCUSSION THREAD

**Discussion by bisejdiu** [molstar/molstar#113] [Q&A] [answered]
I would like to change how the sequences of polymers are shown on the sequence panel. Specifically, instead of having a select box for choosing structures, I want all loaded polymer sequences to be displayed at once. For simplicity, let's say all loaded structures contain only proteins and nothing else. 

What I tried is, I changed the css code so it doesn't break the sequences, and inside `SequenceView` I call the following (i.e. multiple times): 
```javascript
 {<Sequence sequenceWrapper={sequenceWrapper} />}
 {<Sequence sequenceWrapper={sequenceWrapper} />}
```
And I do end up with two parallel sequences inside the `msp-sequence` container, but only the last one works, and I don't know if this would work for sequences of different proteins. 

So, what would be the best way to do this? I'm thinking of rewriting the `Sequence` class in *mul-plugin-ui/sequence/sequence.tsx*, but perhaps there's a better way that uses/repurposes existing code?

**Comment by dsehnal** [comment]
That's a good feature request. I will have a look into adding a select to show all polymers, all chains, or a single chain during the week.

**Reply by bisejdiu** [reply]
Fantastic! That's a feature that would be really useful to me. Looking forward to hearing updates on this.

**Comment by dsehnal** [comment, marked as answer]
Added the option to display all polymers or chains at the same time and optimized the control itself a bit.

![image](https://user-images.githubusercontent.com/21145776/103295027-6678b780-49f3-11eb-89d0-11ef26ffde4f.png)

Do you need NPM package as well?

**Reply by bisejdiu** [reply]
This is great and it's going to be really helpful. Thank you, @dsehnal!

**Reply by dsehnal** [reply]
There was a bug in the polymers/everything mode where the highlight was offset by 1. Hopefully fixed now.
